src/FilterList.js

```diff
@@ -2,16 +2,84 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import {ResultPropTypes} from './ResultPropTypes'
 
-const FilterList = ({filteredResults, resultsMessage, ResultsHeaderClass, ResultElementClass}) => {
-  const filteredElements = filteredResults.map((result) => result.element)
-
-  return (
-    <div>
-      <h4>{resultsMessage}</h4>
-      { filteredElements.length && ResultsHeaderClass && <ResultsHeaderClass /> }
-      { filteredElements.map((element, index) => <div key={index}><ResultElementClass {...element}/></div>) }
-    </div>
+function dynamicSort(property, sortState) {
+  const sortOrder = sortState ?  -1 : 1
+  return function (a,b) {
+    let result = (a[property] < b[property]) ? -1 : (a[property] > b[property]) ? 1 : 0
+    return result * sortOrder
+  }
+}
+
+function titleCase(str) {
+  const splitStr = str.toLowerCase().split(` `).map((string, index)=>
+    index === 0 ?
+      string.charAt(0).toLowerCase() + string.substring(1) :
+      string.charAt(0).toUpperCase() + string.substring(1)    
   )
+  return splitStr.join(``)
+}
+
+class FilterList extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      sortTitle: `species`,
+      sortState: true
+    }
+    this.constructTableHeader = this.constructTableHeader.bind(this)
+    this.sortTable = this.sortTable.bind(this)
+  }
+  
+  constructTableHeader(ResultsHeaderClass){
+    const tableTitles = Object.keys(ResultsHeaderClass().titles)
+    const tableTitleDivs = Object.values(ResultsHeaderClass().titles)
+    const ContainerDiv = ResultsHeaderClass().container
+
+    const TableHeader = () => 
+      <ContainerDiv>
+        { 
+          tableTitles.map((title, index) => {
+            const TitleDiv = tableTitleDivs[index]
+            return Object.keys(this.props.filteredResults[0].element).includes(titleCase(title)) ?
+              titleCase(title) === this.state.sortTitle ?
+                this.state.sortState ?
+                  <TitleDiv key={title} style={{opacity: 1}}> <p id={`selected`} onClick={this.sortTable}>{title} &darr;</p></TitleDiv> :
+                  <TitleDiv key={title} style={{opacity: 1}}> <p id={`selected`} onClick={this.sortTable}>{title} &uarr;</p></TitleDiv> 
+                :
+                <TitleDiv key={title}> <p id={`title`} onClick={this.sortTable}>{title} &uarr;</p></TitleDiv> 
```

Comment:
You are using an HTML entity for the upwards-pointing arrow and a Unicode escape sequence below for the downwards-pointing arrow. Use one or the other (I’d suggest you just copy and paste the character, since we can write code in UTF-8 now). I’d however use triangles, since those are most widely used in other tables: ▲, ▼.

Reply:
Excellent point!